File: Challenges/4kyu/strip_comments/python/solution.py
def strip_comments(strng, markers):
    strng = strng.split("\n")

    for i in range (len(strng)):
        line = ""

        for j in range (len(strng[i])):
            if (strng[i][j] in markers): break
            line += strng[i][j]
    
        strng[i] = line.rstrip()
        
    return "\n".join(strng)

File: Challenges/4kyu/strip_comments/javascript/solution.js
function solution(text, markers) {
  text = text.split("\n")
  
  for (let i = 0; i < text.length; i++) {
    let line = ""
    
    for (let j = 0; j < text[i].length; j++) {
      if (markers.includes(text[i][j])) break
      line += text[i][j]
    }
    
    text[i] = line.trimEnd()
  }
  
  return text.join("\n")
}